src/js/brim/program.ts

```diff
@@ -1,39 +1,33 @@
 import {parseAst, zed} from "@brimdata/zealot"
-import {isEmpty} from "lodash"
+import {isEmpty, last} from "lodash"
 import {trim} from "../lib/Str"
-import stdlib from "../stdlib"
-import brim from "./"
-import {ANALYTIC_PROCS} from "./ast"
-
-export default function (p = "", pins: string[] = []) {
-  p = concatPins(p, pins)
+import ast, {ANALYTIC_PROCS} from "./ast"
+import syntax from "./syntax"
 
+export default function (p = "") {
```

Comment:
Do you think this `program` logic should eventually be moved nearer/inside the `BrimQuery` code? They seem to have similar sets of knowledge for query/program manipulation.

Reply:
Yep, that feels right.